src/formpack/reporting/export.py

```diff
@@ -401,38 +398,6 @@ def _get_attachment(val, field, attachments):
                 if re.match(fr'^.*/{_val}$', f['filename']) is not None
             ]
 
-        def _get_value_from_supplemental_details(
-            field: FormField, supplemental_details: Dict
-        ) -> Optional[str]:
-            source, name = field.analysis_path
-            _sup_details = supplemental_details.get(source, {})
-
-            if not _sup_details:
-                return
-
-            # The names for translation and transcript fields are in the format
-            # of `translated_<language code>` which must be stripped to get the
-            # value from the supplemental details dict
-            if _name := re.match(r'^(translation|transcript)_', name):
-                name = _name.groups()[0]
-
-            val = _sup_details.get(name)
-            if val is None:
-                return ''
-
-            return val
-
-        def _get_value_from_entry(
-            entry: Dict, field: FormField, supplemental_details: Dict
-        ) -> Optional[str]:
-            if field.analysis_question and supplemental_details:
-                return _get_value_from_supplemental_details(
-                    field, supplemental_details
-                )
-
-            suffix = 'meta/' if field.data_type == 'audit' else ''
-            return entry.get(f'{suffix}{field.path}')
```

Comment:
Let's do proper object-oriented programming and let fields be responsible for handling their own business